src/components/LocationWeather/LocationWeather.jsx

```diff
@@ -0,0 +1,85 @@
+import './LocationWeather.css';
+import React, { Component, PureComponent } from 'react';
+import {
+  API_AUTH_PASS,
+  API_AUTH_USERNAME,
+  API_FORECAST_3_HOURLY,
+  API_FORECAST_3_HOURLY_PERIODS,
+  API_FORECAST_DAILY,
+  API_KIEV_ID
+} from '../../constants/constants';
+import { weatherAPI } from '../../services/dataService';
+import LocationWeatherCurrent from './LocationWeatherCurrent/LocationWeatherCurrent';
+import LocationWeatherDailyList from './LocationWeatherDailyList/LocationWeatherDailyList';
+import LocationWeather3HourlyList from './LocationWeather3HourlyList/LocationWeather3HourlyList';
+
+class LocationWeather extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentLocationId: API_KIEV_ID,
+      currentLocationInfo: null,
+      currentLocationWeather: null,
+      currentLocationDailyWeather: null,
+      currentLocation3HourlyWeather: null,
+      activeDayDate: ''
+    };
+
+    this.setActiveDayDate = this.setActiveDayDate.bind(this);
+  }
+
+  componentDidMount() {
+    (async () => {
+      try {
+        await weatherAPI.getToken(API_AUTH_USERNAME, API_AUTH_PASS);
+        const currentLocationWeather = await weatherAPI.getCurrentWeather(API_KIEV_ID);
+        this.setState({
+          currentLocationWeather,
+          activeDayDate: currentLocationWeather.time.split('T')[0]
+        });
+        const currentLocationInfo = await weatherAPI.getLocationInfo(API_KIEV_ID);
+        this.setState({ currentLocationInfo });
+        const currentLocationDailyWeather = await weatherAPI.getForecast(
+          API_FORECAST_DAILY,
+          API_KIEV_ID
+        );
+        this.setState({ currentLocationDailyWeather });
+        const currentLocation3HourlyWeather = await weatherAPI.getForecast(
+          API_FORECAST_3_HOURLY,
+          API_KIEV_ID,
+          API_FORECAST_3_HOURLY_PERIODS
+        );
+        this.setState({ currentLocation3HourlyWeather });
+        console.log(currentLocation3HourlyWeather);
+      } catch (error) {
+        console.log(error);
+      }
+    })();
+  }
+
+  setActiveDayDate(date) {
+    this.setState({ activeDayDate: date });
+  }
+
+  render() {
+    return (
+      <div className="location-weather">
+        <LocationWeatherCurrent
+          currentLocationWeather={this.state.currentLocationWeather}
+          currentLocationInfo={this.state.currentLocationInfo}
+        />
+        <LocationWeatherDailyList
+          currentLocationDailyWeather={this.state.currentLocationDailyWeather}
+          activeDayDate={this.state.activeDayDate}
+          setActiveDayDate={this.setActiveDayDate}
+        />
+        <LocationWeather3HourlyList
```

Comment:
I feel iffy about these 3 hours thing. Could it be parametrized instead? Could you have
```
<LocationWeatherHourlyList length={3} />
```
something like that

Or in the worst case scenario just omit 3 in the name. It would still make it hourlyforecast. It is just your app displays only 3 hours.


That applies to the whole PR

Reply:
I renamed all the "3_hourly" things to "detailed". I refused using "hourly" because the interval can be less than a hour. In case of change needed interval I'll have to change only 2 things - API_FORECAST_DETAILED endpoint string and dataFilter function inside LocationWeatherDetailedList component.